src/object/srv_obj_migrate.c

```diff
@@ -2013,11 +2013,11 @@ migrate_one_ult(void *arg)
 	D_DEBUG(DB_REBUILD, "mrone %p inflight_size "DF_U64" max "DF_U64"\n",
 		mrone, tls->mpt_inflight_size, tls->mpt_inflight_max_size);
 
-	while (tls->mpt_inflight_size + data_size >=
-	       tls->mpt_inflight_max_size && tls->mpt_inflight_max_size != 0
-	       && !tls->mpt_fini) {
-		D_DEBUG(DB_REBUILD, "mrone %p wait "DF_U64"/"DF_U64"\n", mrone,
-			tls->mpt_inflight_size, tls->mpt_inflight_max_size);
+	while (tls->mpt_inflight_size + data_size >= tls->mpt_inflight_max_size &&
```

Comment:
An idea, possibly for a future patch:
would it be helpful to emit a warning log message when mpt_inflight_size == 0 and when data_size exceeds tls->mpt_inflight_max_size? To show that the code is allowing the migrate to happen while exceeding the maximum amount of in-flight data? In some cases could we be surprised to see a very large data_size? And maybe with PR [14059](https://github.com/daos-stack/daos/pull/14059 ) the warning could be a hint to adjust higher an engine environment variable?

Reply:
sure, probably can add a WARN msg in following patch later.